src/components/authentication/authentication.service.ts

```diff
@@ -256,11 +258,44 @@ export class AuthenticationService {
   }
 
   async changePassword(
-    _oldPassword: string,
-    _newPassword: string,
-    _session: ISession
+    oldPassword: string,
+    newPassword: string,
+    session: ISession
   ): Promise<void> {
-    // TODO
+    const result = await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([
+        node('requestingUser'),
+        relation('out', '', 'password', { active: true }),
+        node('password', 'Property', { active: true }),
+      ])
+      .return('password.value as passwordHash')
+      .first();
+
+    if (!result || !(await argon2.verify(result.passwordHash, oldPassword))) {
+      throw new UnauthenticatedException('Invalid credentials');
+    }
+    if (oldPassword === newPassword) {
+      throw new NotAcceptableException(
+        'newPassword could not be same with oldPassword'
+      );
+    }
```

Comment:
Let's skip this for now. We don't have a good reason to block them changing their password to what it is currently. Esp since they just confirmed they know their old password.

Reply:
Okay